scipy/stats/stats.py

```diff
@@ -2821,6 +2821,25 @@ def iqr(x, axis=None, rng=(25, 75), scale=1.0, nan_policy='propagate',
     return out
 
 
+def _mad_1d(x, center, nan_policy):
+    # Median absolute deviation for 1-d array x.  No warning is generated
+    # if x is empty or all nan.
+    # x must be a 1-d numpy array.
+    # If nan_policy is not 'propagate', it is assumed to be 'omit'.
+    isnan = np.isnan(x)
+    if isnan.any():
+        if nan_policy == 'propagate':
+            return np.nan
+        x = x[~isnan]
+    if x.size == 0:
+        # MAD of an empty array is nan.
+        return np.nan
+    # Edge cases have been handled, so do the basic MAD calculation.
+    med = center(x)
```

Comment:
I realize this is a 'private' function so we wouldn't want to check that `center` argument is indeed callable, correct? 

I noticed that the `median_abs_deviation` does ***not*** do a check on `callable(center)` but the docstring indicates is optional and supports callable.

If we don't want here I can open a separate PR to fix that issue in `median_abs_deviation` to not delay this one merging?

Reply:
Right, this helper function should only be called from `median_abs_deviation`, so it shouldn't need to validate its arguments.

Reply:
see updated comments in latest, looks good, thanks.